front/src/common/Map/WarpedMap/core/quadtree.ts

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable prefer-destructuring, no-plusplus */
+import { BBox2d } from '@turf/helpers/dist/js/lib/geojson';
+import { Feature, FeatureCollection, GeoJsonProperties, Geometry, Position } from 'geojson';
+import bbox from '@turf/bbox';
+
+// The following types help describing a full QuadTree:
+export type Leaf<T> = {
+  type: 'leaf';
+  elements: T[];
+};
+export type Quad<T> = {
+  type: 'quad';
+  bbox: BBox2d;
+  children: [QuadChild<T> | null, QuadChild<T> | null, QuadChild<T> | null, QuadChild<T> | null];
+};
+export type QuadChild<T> = Quad<T> | Leaf<T>;
+
+export function bboxIntersect([mx1, my1, Mx1, My1]: BBox2d, [mx2, my2, Mx2, My2]: BBox2d): boolean {
+  return !(mx1 > Mx2) && !(Mx1 < mx2) && !(my1 > My2) && !(My1 < my2);
+}
+
+export function getNewQuadChild<T>(box: BBox2d, isLeaf?: boolean): QuadChild<T> {
+  return isLeaf
+    ? {
+        type: 'leaf',
+        elements: [],
+      }
+    : {
+        type: 'quad',
+        bbox: box,
+        children: [null, null, null, null],
+      };
+}
+
+/**
+ * This function takes a collection of GeoJSON features and a depth, and returns a QuadTree of the given depth, with all
+ * features properly indexed.
+ */
+export function getQuadTree<G extends Geometry | null = Geometry, P = GeoJsonProperties>(
+  collection: FeatureCollection<G, P>,
+  depth: number
+): Quad<Feature<G, P>> {
+  const boundingBox = bbox(collection) as BBox2d;
+  const root = getNewQuadChild(boundingBox) as Quad<Feature<G, P>>;
+
+  for (let i = 0, l = collection.features.length; i < l; i++) {
+    const feature = collection.features[i];
+    const fBBox = bbox(feature) as BBox2d;
+
+    let quads: QuadChild<Feature<G, P>>[] = [root];
+    for (let d = 0; d < depth; d++) {
+      if (!quads.length) break;
+
+      const newQuads: QuadChild<Feature<G, P>>[] = [];
+      for (let j = 0, quadsCount = quads.length; j < quadsCount; j++) {
```

Comment:
Here, your version does more lookups. It might seem like nothing, but in this `getQuadTree` function, it can be called thousands of times, and I tried to get the code as efficient as I could:
- All loops are vanilla `for (...;...;...)` loops (faster)
- I try to cache everything I can, including `quadsCount`

I think it's better, and I don't see the problem with it.

Reply:
OK, it was only a suggestion, I didn't know what was the reason behind this variable instantiation